boa/src/syntax/lexer/mod.rs

```diff
@@ -469,9 +474,19 @@ impl<'a> Lexer<'a> {
                             _ => break,
                         }
                     }
-                    // TODO make this a bit more safe -------------------------------VVVV
+
                     self.push_token(TokenKind::NumericLiteral(
-                        f64::from_str(&buf).map_err(|_| LexerError::new("Could not convert value to f64"))?,
+                        if is_floating_point {
+                            // TODO make this a bit more safe -------------------------------VVVV
+                            NumericLiteral::Number(
+                                f64::from_str(&buf).map_err(|_| LexerError::new("Could not convert value to f64"))?
+                            )
+                        } else {
+                            NumericLiteral::Integer(
+                                u64::from_str(&buf).map_err(|_e| {dbg!(&_e); dbg!(&is_floating_point); LexerError::new("Could not convert value to u64")})?
```

Comment:
The `dbg!()` macro should be removed before merging.
```suggestion
                                u64::from_str(&buf).map_err(|_e| LexerError::new("Could not convert value to u64"))?
```